tests/rptest/services/openmessaging_benchmark.py

```diff
@@ -118,77 +119,60 @@ class OpenMessagingBenchmark(Service):
     DRIVER_FILE = os.path.join(OPENMESSAGING_DIR,
                                "driver-redpanda/redpanda-ducktape.yaml")
     WORKLOAD_FILE = os.path.join(OPENMESSAGING_DIR, "workloads/ducktape.yaml")
-    OUTPUT_FILE = os.path.join(PERSISTENT_ROOT, "result.json")
+    NUM_WORKERS = 2
 
     logs = {
         "open_messaging_benchmark_stdout_stderr": {
             "path": STDOUT_STDERR_CAPTURE,
             "collect_default": True
         },
         "open_messaging_benchmark_result": {
-            "path": OUTPUT_FILE,
             "collect_default": True
         }
     }
 
-    def __init__(self, ctx, redpanda):
+    def __init__(self, ctx, redpanda, driver_idx=0, workload_idx=0):
+        """
+        Creates a utility that can run OpenMessagingBenchmark (OMB) tests in ducktape. See OMB
+        documentation for definitions of driver/workload files.
+        """
         super(OpenMessagingBenchmark, self).__init__(ctx, num_nodes=1)
         self._ctx = ctx
         self.redpanda = redpanda
-        self.set_default_configuration()
         self.workers = None
-
-    def set_default_configuration(self):
-        self.configuration = {
-            "topics": 1,
-            "partitions_per_topic": 3,
-            "subscriptions_per_topic": 1,
-            "producers_per_topic": 1,
-            "consumer_per_subscription": 1,
-            "consumer_backlog_size_GB": 0,
-        }
+        self.driver = OMBSampleConfigurations.DRIVERS[driver_idx]
+        self.workload = OMBSampleConfigurations.WORKLOADS[workload_idx]
+        self.set_env_based_overrides(driver_idx, workload_idx)
+        output_file_name = "result-{}-{}.json".format(self.driver["name"],
+                                                      self.workload["name"])
+        self.output_file = os.path.join(OpenMessagingBenchmark.PERSISTENT_ROOT,
+                                        output_file_name)
+        self.logs["open_messaging_benchmark_result"]["path"] = self.output_file
+        self.logger.info("Using driver: %s, workload: %s", self.driver["name"],
+                         self.workload["name"])
+
+    def set_env_based_overrides(self, driver_idx, workload_idx):
+        """Default configuration to use based on the environment in which the driver is running."""
+        # We pick different defaults depending on the environment we are in. Dedicated nodes get a heavier
+        # workload by default if sample is picked is picked.
         if self.redpanda.dedicated_nodes:
```

Comment:
This looks weird. I understand previously `dedicated_nodes` used to adjust duration of the tests. So the existing 4 tests:

- test_multiple_subscriptions_omb
- test_multiple_producers_consumers_omb
- test_multiple_topics_omb
- test_default_omb_configuration

were adjusted according to the environment.

Not OpenMessagingBenchmark is parametrized and dedicated_nodes replaces the whole template. It's like as in the previous version `test_multiple_topics_omb` was silently replaced with `test_default_omb_configuration`. It's very implicit and may lead to unexpected behavior.

For example let's took this test

```
@matrix(driver_idx=[0, 1], workload_idx=[0])
def test_default_omb_configuration(self, driver_idx, workload_idx)
```

The matrix parameter expands into `[[0,0],[1,0]]` but `dedicated_nodes` is true it is being replaced with `[[0,0],[0,0]]` and we execute the same test twice withing it does something different while it isn't true.

It may be better if we have solved this problem at higher level. E.g. we could have

  - openmessaging_benchmark_cdt_test.py
  - openmessaging_benchmark_local_test.py
  - openmessaging_benchmark_etc_test.py

files and just include / exclude them for different suites